### DISCUSSION TITLE
Fluent mapping for CreateTable

### DISCUSSION THREAD

**Discussion by usdaud** [linq2db/linq2db#3704] [Q&A] [answered]
I've did fluent mapping for my freshy creating entity MyClass1 (contains dosen of properties but for DB need use only few of them) like that

```
var builder = db.MappingSchema.GetFluentMappingBuilder();

builder
	.Entity<MyClass1>()
	.Property(m => m.Field1).HasLength(256)
	.Property(m => m.Field2).HasLength(256)
	.Property(m => m.Field3)
;
```

but when I try execute create table statement 

`table = db.CreateTable<MyClass1>();`

it is try create table sql query with all columns.

Fluent mapping do not compatible with creating table?

**Comment by MaceWindu** [comment, marked as answer]
by default all properties recognized as columns. To disable this behavior add call to `IsColumnRequired()` method

**Reply by usdaud** [reply]
Starting from 5.0.0 version this doesn't work anymore.

```
private class TestEntity
{
	public string Column1 { get; set; }
	public string Column2 { get; set; }
	public string Column3 { get; set; }
}

var builder = new FluentMappingBuilder();

builder
	.Entity<TestEntity>()
	.IsColumnRequired()
	.Property(m => m.Column1)
	.Property(m => m.Column2)
	.Build()
	;

table = db.CreateTable<TestEntity>();
```

In Sql server table creates with 3 columns.

**Reply by MaceWindu** [reply]
Do you pass mapping schema with configured fluent mappings to context?

**Reply by usdaud** [reply]
My bad.

**Comment by usdaud** [comment]
Seems it works, thanks. But anyway stuck on insert operation.

`db.Insert(myObj1);`

throws cast exception. In debug mode I still see in SqlTable.ctor enumeration of all properties, not just pointed by fluent mapper only.

**Reply by MaceWindu** [reply]
I need more details. Mapping, exception...
